CD4PE.md

```diff
@@ -0,0 +1,327 @@
+export BEAKER_PE_VER=2019.0.1
+```
+
+or use the provided environment setup file:
+```
+source config/env/env_setup_2019.0.1
+```
+
+### Provision GPLT hosts and run the pre-suite
+Perform a 'performance' run to provision the 'mom' and 'metrics' hosts via ABS and run the Beaker pre-suite.
+For a standard performance test environment run the `performance_setup` rake task.
+For a scale test environment run the `autoscale_setup` rake task.
+
+### Provision the CD4PE hosts
+The following rake tasks are provided to create the CD4PE test environment hosts via ABS:
+```
+rake abs_provision_environment_cd4pe          # Provision the hosts for a cd4pe test environment via ABS
+rake abs_provision_host_agent                 # Provision an agent host via ABS
+rake abs_provision_host_cd4pe                 # Provision a cd4pe host via ABS
+rake abs_provision_host_gitlab                # Provision a gitlab host via ABS
+rake abs_provision_host_worker                # Provision a worker host via ABS
+
+```
+
+For a full CD4PE environment run the `abs_provision_environment_cd4pe` task.
+For a custom environment, run the individual tasks to provision the desired hosts.
+
+## Set up Gitlab
+The test environment uses a standard installation of Gitlab.
+
+### Install the Gitlab package
+* SSH to the Gitlab host as the root user.
+* Add the GitLab package repository.
+
+```
+curl https://packages.gitlab.com/install/repositories/gitlab/gitlab-ee/script.rpm.sh | sudo bash
+
+
+```
+
+* Install the Gitlab package (replace 'ip-x-x-x-x.amz-dev.puppet.net' with the value for your Gitlab host)
+```
+EXTERNAL_URL="http://ip-x-x-x-x.amz-dev.puppet.net" yum install -y gitlab-ee
+```
+
+### Set the root password
+Once Gitlab is installed and running, navigate to the host in a browser.
+* Set the password for the 'root' account to 'puppetlabs'.
+* Navigate to `admin/application_settings/network` and enable the 'Allow requests to the local network from hooks and services' setting.
+
+### Import the control repo
+In the Gitlab UI:
+* Navigate to New Project → Import Project → Repo by URL
+* Provide a project name and description
+* Provide the control repo URL: https://github.com/puppetlabs/puppetlabs-puppetserver_perf_control.git
+* Click the 'Create project' button
+
+You should now see the control repo in Gitlab.
+* Disable the 'auto-dev-ops' pipeline
+
+### Update the control repo for CD4PE
+CD4PE is designed to use the master branch as the default with additional branches per environment.
+
+* Create a 'master' branch from the 'production' branch.
+* Set the 'master' branch as the default.
+* Create additional branches per environment.
+* Remove extraneous branches.
+* Ensure all branches are unprotected.
+
+# Set up PE
+## Enable Code Manager
+We'll be using code manager to deploy the cd4pe module, so we'll set it up before proceeding based on [the docs](https://puppet.com/docs/pe/2019.0/code_mgr_config.html#enable-code-manager-after-installation)
```

Comment:
When I read this i thought I was supposed to go to the doc and do what it says.  But really you are giving the link for reference or if the steps below don't work as expected...

Reply:
Updated